src/tests/test_libs_xml_utils.py

```diff
@@ -244,25 +269,29 @@ def test_bad_cert_included(self) -> None:
         )
 
     def test_fail_replaced_cert(self) -> None:
+        """
+        Tests that the signature verification fails
+        when the certificate is not the one that was used to sign the document.
+        """
         xml_doc = parse_untrusted_xml(self.with_replaced_cert)
-        cert = load_pem_x509_cert(self.any_x509_cert_pem_file)
+        cert = load_pem_x509_cert(self.xml_doc_cert_pem_bytes)
 
         with self.assertRaises(XmlSignatureInvalid) as cm:
             verify_xml_signature(xml_doc, trusted_x509_cert=cert)
         self.assertEqual(
             cm.exception.args,
-            ("Signature verification failed: []",),
+            ("Signature verification failed: ",),
         )
 
     def test_fail_included_cert_not_from_a_known_ca(self) -> None:
-        xml_doc = parse_untrusted_xml(self.with_valid_signature)
+        xml_doc = parse_untrusted_xml(self.with_valid_signature_not_expired)
 
         # Without cert: fails because the issuer of the cert in the signature is not a known CA.
         with self.assertRaises(XmlSignatureInvalidCertificate) as cm:
             verify_xml_signature(xml_doc, trusted_x509_cert=None)
         self.assertEqual(
             cm.exception.args,
-            ('unable to get local issuer certificate',),
+            ('validation failed: Certificate is missing required extension',),
         )
```

Comment:
Ref: https://github.com/cordada/lib-cl-sii-python/pull/736#discussion_r1880905930

```python
import datetime
import unittest.mock
from typing import Any

import signxml.verifier

from cl_sii.base.constants import SII_OFFICIAL_TZ
from cl_sii.libs.tz_utils import convert_naive_dt_to_tz_aware


def test_fail_included_cert_not_from_a_known_ca(self) -> None:
    xml_doc = parse_untrusted_xml(self.with_valid_signature)
    xml_doc_signature_timestamp = convert_naive_dt_to_tz_aware(
        dt=datetime.datetime.fromisoformat('2019-04-01T01:36:40'),  # From XML doc’s <TmstFirma>
        tz=SII_OFFICIAL_TZ,
    )

    def _get_cert_chain_verifier(
        *args: Any, **kwargs: Any
    ) -> signxml.util.X509CertChainVerifier:
        # The default signature verification time is the current time (see
        # https://cryptography.io/en/43.0.3/x509/verification/#cryptography.x509.verification.PolicyBuilder.time
        # ), but that causes verification to fail with the message
        # “validation failed: cert is not valid at validation time”.
        # To avoid that, we set the verification time to the time of the signature.
        return signxml.util.X509CertChainVerifier(
            ca_pem_file=kwargs['ca_pem_file'], verification_time=xml_doc_signature_timestamp
        )

    # Without cert: fails because the issuer of the cert in the signature is not a known CA.
    with self.assertRaises(XmlSignatureInvalidCertificate) as cm, unittest.mock.patch.object(
        signxml.verifier.XMLVerifier,
        'get_cert_chain_verifier',
        side_effect=_get_cert_chain_verifier,
    ) as mock_get_cert_chain_verifier:
        verify_xml_signature(xml_doc, trusted_x509_cert=None)
    self.assertEqual(
        cm.exception.args,
        # According to some test cases from https://x509-limbo.com/, OpenSSL’s error message
        # “unable to get local issuer certificate” seems to be equivalent to PyCA Cryptography’s
        # error message below:
        (
            'validation failed:'
            ' candidates exhausted:'
            ' all candidates exhausted with no interior errors',
        ),
    )
    mock_get_cert_chain_verifier.assert_called_once_with(ca_pem_file=None)
```